smudge-api.el

```diff
@@ -11,84 +11,79 @@
  particular language."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-api-country "US"
+(defcustom smudge-api-country "US"
   "Optional.  An ISO 3166-1 alpha-2 country code.
 Provide this parameter if you want to narrow the list of returned categories
 to those to a particular country.  If omitted, the returned items will be
 globally relevant."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-oauth2-callback-port "8080"
+(defcustom smudge-oauth2-callback-port "8080"
   "The port for the httpd to listen on for the OAuth2 callback."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-oauth2-callback-endpoint "/spotify-callback"
+(defcustom smudge-oauth2-callback-endpoint "/smudge-callback"
   "The endpoint for the httpd to listen on for the OAuth2 callback."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defconst spotify-api-endpoint     "https://api.spotify.com/v1")
-(defconst spotify-oauth2-auth-url  "https://accounts.spotify.com/authorize")
-(defconst spotify-oauth2-token-url "https://accounts.spotify.com/api/token")
-(defconst spotify-oauth2-scopes    "playlist-read-private playlist-read-collaborative playlist-modify-public playlist-modify-private user-read-private user-read-playback-state user-modify-playback-state user-read-playback-state user-read-recently-played")
-(defconst spotify-oauth2-callback  (concat "http://localhost:" spotify-oauth2-callback-port "/spotify-callback"))
+(declare-function oauth2-request-access "oauth2")
+(declare-function oauth2-refresh-access "oauth2")
 
-(defun spotify-httpd-stop ()
+(defvar *smudge-user*         nil
+  "Cached user object.")
+(defvar *smudge-api-oauth2-token* nil
+  "Cached OAuth2 token.")
+(defvar *smudge-api-oauth2-ts*    nil
+  "Unix timestamp in which the OAuth2 token was retrieved.
+This is used to manually refresh the token when it's about to expire.")
+(defvar *smudge-api-oauth2-token-directory* (concat (file-name-as-directory user-emacs-directory) ".cache/smudge")
+	"Directory where the OAuth2 token is serialized.")
+(defvar *smudge-api-oauth2-token-file* (concat *smudge-api-oauth2-token-directory* "/" "token")
+	"Location where the OAuth2 token is serialized.")
+
+(defvar *smudge-is-authorizing* nil
+	"Whether smudge is in the process of obtaining an OAuth2 token.")
+
+(defconst smudge-api-endpoint     "https://api.spotify.com/v1")
+(defconst smudge-api-oauth2-auth-url  "https://accounts.spotify.com/authorize")
+(defconst smudge-api-oauth2-token-url "https://accounts.spotify.com/api/token")
+(defconst smudge-api-oauth2-scopes    "playlist-read-private playlist-read-collaborative playlist-modify-public playlist-modify-private user-read-private user-read-playback-state user-modify-playback-state user-read-playback-state user-read-recently-played")
+(defconst smudge-api-oauth2-callback  (concat "http://localhost:" smudge-oauth2-callback-port "/smudge-api-callback"))
```

Comment:
The README is documenting a different value for the callback, `smudge-callback`. We need to make the default value the same in both places.

Reply:
@danielfm Fixed in README.md!